### PR TITLE
Add 0.17 release notes for milestone 1

### PR DESCRIPTION
**Pull request by SueChaplain**
New directory and release notes entry for 0.17.0

On the 0.16.0, uncommented some lines to show final release information,
which had been hidden during milestone builds.

[ci-skip]

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

File: doc/release-notes/0.17/0.17.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+<td valign="top">Provides the ability to enable and disable the storing and finding of Unsafe classes in the shared classes cache, which are created with <tt>Unsafe.defineClass</tt>. This option is enabled by default.</td>
+</tr>
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7089">#7089</a></td>
+<td valign="top">New option <tt>-XX:[+|-]ClassRelationshipVerifier</tt> </td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">Provides the ability to enable and disable the recording and validating of class relationships in the verifier. When disabled, unnecessary class loading is avoided, which can reduce VM startup time. This option cannot be used with <tt>-Xfuture</tt>, which is also enabled when the <tt>-Xverify:all</tt> option is set.</td>
+</tr>
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7089">#7089</a></td>
```
Comment (pshipton) [inline doc/release-notes/0.17/0.17.md]: Wrong pull request here

File: doc/release-notes/0.17/0.17.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7149">#7149</a></td>
+<td valign="top">New option <tt>-XX:[+|-]ShareUnsafeClasses</tt> </td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">Provides the ability to enable and disable the storing and finding of Unsafe classes in the shared classes cache, which are created with <tt>Unsafe.defineClass</tt>. This option is enabled by default.</td>
+</tr>
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7089">#7089</a></td>
+<td valign="top">New option <tt>-XX:[+|-]ClassRelationshipVerifier</tt> </td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">Provides the ability to enable and disable the recording and validating of class relationships in the verifier. When disabled, unnecessary class loading is avoided, which can reduce VM startup time. This option cannot be used with <tt>-Xfuture</tt>, which is also enabled when the <tt>-Xverify:all</tt> option is set.</td>
```
Comment (pshipton) [inline doc/release-notes/0.17/0.17.md]: I think should should say "When enabled, ...", with the default being disabled.

File: doc/release-notes/0.17/0.17.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+<th valign="bottom">Issue number</th>
+<th valign="bottom">Description</th>
+<th valign="bottom">Version / Platform</th>
+<th valign="bottom">Impact</th>
+</tr>
+</thead>
+<tbody>
+
+
+tr><td valign="top"><a href="https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/324">#324</a> <a href="https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/191">#191</a></td>
```
Comment (pshipton) [inline doc/release-notes/0.17/0.17.md]: Missing `<` before `tr>`

File: doc/release-notes/0.17/0.17.md (hunk 1 above)
Comment (SueChaplain) [reply doc/release-notes/0.17/0.17.md]: Changing to https://github.com/eclipse/openj9/pull/7213

File: doc/release-notes/0.17/0.17.md (hunk 2 above)
Comment (SueChaplain) [reply doc/release-notes/0.17/0.17.md]: OK

File: doc/release-notes/0.17/0.17.md (hunk 3 above)
Comment (SueChaplain) [reply doc/release-notes/0.17/0.17.md]: Well spotted. My editor fixed the preview so I didn't notice.

**Comment by pshipton** [comment]
@gita-omr @andrewcraik @fjeremic @DanHeidinga 
anything missing or any other review comments before this is merged?

**Comment by pshipton** [comment]
What about https://github.com/eclipse/openj9-docs/issues/366 - Support for the z15 processor ?

**Comment by pshipton** [comment]
We might mention https://github.com/eclipse/openj9-docs/issues/359 - `recent builds from adoptopenjdk.net are CUDA-enabled` as well

**Comment by pshipton** [comment]
Did https://github.com/eclipse/openj9-docs/issues/333 - jcmd get documented?

**Comment by pshipton** [comment]
jcmd is in the 0.16 release notes. Maybe it's worth referencing the 0.16 release notes from the 0.17 release notes. People using jdk8 or jdk11 will be moving from 0.15 to 0.17 and might miss the list of things in the 0.16 release.

**Comment by SueChaplain** [comment]
> jcmd is in the 0.16 release notes. Maybe it's worth referencing the 0.16 release notes from the 0.17 release notes. People using jdk8 or jdk11 will be moving from 0.15 to 0.17 and might miss the list of things in the 0.16 release.

Yes - I've never been very comfortable with how the releases get reflected in the docs. We get confused, so for sure our users do too. Not sure whether there is a better approach.

**Comment by SueChaplain** [comment]
> We might mention eclipse/openj9-docs#359 - recent builds from adoptopenjdk.net are CUDA-enabled as well

This should really be in the AdoptOpenJDK release notes, not OpenJ9 release notes. We do have an AdoptOpenJDK topic in the user doc, where I can mention it.

**Comment by SueChaplain** [comment]
> jcmd is in the 0.16 release notes. Maybe it's worth referencing the 0.16 release notes from the 0.17 release notes. People using jdk8 or jdk11 will be moving from 0.15 to 0.17 and might miss the list of things in the 0.16 release.

We'll do this in the user doc, where I think it makes more sense.

File: doc/release-notes/0.17/0.17.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7213">#7213</a></td>
+<td valign="top">New experimental <tt>-Xshareclasses</tt> sub-options for Docker containers</td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">The <tt>-Xshareclasses:createLayer</tt> and <tt>-Xshareclasses:layer=<number></tt> options are intended
+for users who are running Java applications in docker containers. These options allow the shared classes cache to be
+built into different layers of a docker image. These options are experimental.</td>
+</tr>
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7213">#7213</a></td>
```
Comment (fjeremic) [inline doc/release-notes/0.17/0.17.md]: This looks like the wrong issue number. I think it should be #5884.

File: doc/release-notes/0.17/0.17.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+<td valign="top">Support for the IBM z15 processor.</td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">This release of OpenJ9 supports the z15 processor.</td>
```
Comment (fjeremic) [inline doc/release-notes/0.17/0.17.md]: These lines seem to imply that earlier releases won't work on z15, which is definitely not true. I think we should phrase this more along the lines of "This release adds JIT compiler support for exploiting z15 instructions.".

File: doc/release-notes/0.17/0.17.md (hunk 4 above)
Comment (SueChaplain) [reply doc/release-notes/0.17/0.17.md]: Sorry - copied from the line above. Now changed.

File: doc/release-notes/0.17/0.17.md (hunk 5 above)
Comment (SueChaplain) [reply doc/release-notes/0.17/0.17.md]: OK

File: doc/release-notes/0.17/0.17.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7149">#7149</a></td>
+<td valign="top">New option <tt>-XX:[+|-]ShareUnsafeClasses</tt> </td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">Provides the ability to enable and disable the storing and finding of Unsafe classes in the shared classes cache, which are created with <tt>Unsafe.defineClass</tt>. This option is enabled by default.</td>
+</tr>
+
+<tr><td valign="top"><a href="https://github.com/eclipse/openj9/pull/7089">#7089</a></td>
+<td valign="top">New option <tt>-XX:[+|-]ClassRelationshipVerifier</tt> </td>
+<td valign="top">OpenJDK8 and later</td>
+<td valign="top">Provides the ability to enable and disable the recording and validating of class relationships in the verifier. When enabled, unnecessary class loading is avoided, which can reduce VM startup time. This option is disabled by default and cannot be used with <tt>-Xfuture</tt>, which is also enabled when the <tt>-Xverify:all</tt> option is set. </td>
```
Comment (DanHeidinga) [inline doc/release-notes/0.17/0.17.md]:
```suggestion
<td valign="top">Provides the ability to enable and disable the recording and lazy validating of class relationships in the verifier. When enabled, unnecessary class loading is avoided, which can reduce VM startup time. This option is disabled by default and cannot be used with <tt>-Xfuture</tt>, which is also enabled when the <tt>-Xverify:all</tt> option is set. </td>
```

**Comment by pshipton** [comment]
No more comments so I'll go ahead and merge. Anything further can be done separately.

**Comment by gita-omr** [comment]
Late response but LGTM.

**Pull request merged**